lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -352,30 +353,48 @@ def add_line_items(xml, level_3_data)
         }
       end
 
-      def add_stored_credential(xml, options)
+      def add_stored_credential(xml, payment_method, options)
+        return unless options[:stored_credential]
```

Comment:
This is not necessary since you already did` if options[:stored_credential]` when calling the method

Reply:
True, though I'd lean toward overeager guard clauses than less; if it gets used in the future it's already there.